src/api/Controllers/UploadController.cs

```diff
@@ -59,9 +60,14 @@ public async Task<ActionResult<int>> UploadFileAsync(int workgroupId, IFormFile
             // Checks if any of the provided attachments has a whitespace in its file name.
             if (attachments?.Any(pdfFile => pdfFile.FileName.Any(char.IsWhiteSpace)) == true) return BadRequest("One or more file name(s) contain a whitespace.");
 
+            // if a lithology file is provided, checks if it is a CSV file.
+            if (lithologyFile != null && !FileTypeChecker.IsCsv(lithologyFile)) return BadRequest("Invalid file type for lithology csv.");
+
             var boreholeImports = ReadBoreholesFromCsv(boreholesFile);
+            var lithologyImports = ReadLithologiesFromCsv(lithologyFile);
 
             ValidateBoreholeImports(workgroupId, boreholeImports, attachments);
+            ValidateLithologyImports(boreholeImports.Select(bhi => bhi.ImportId).ToList(), lithologyImports);
```

Comment:
REQ: Ich denke wir sollten zuerst die Boreholes validieren, bevor wir die Lithologien auslesen. Also:
```csharp
var boreholeImports = ReadBoreholesFromCsv(boreholesFile);
ValidateBoreholeImports(workgroupId, boreholeImports, attachments);

var lithologyImports = ReadLithologiesFromCsv(lithologyFile);
ValidateLithologyImports(boreholeImports.Select(bhi => bhi.ImportId).ToList(), lithologyImports);
```

Was noch zu überlegen wäre: Aktuell wird `ReadLithologiesFromCsv` und `ValidateLithologyImports` in jedem Fall aufgerufen, auch wenn keine Litho angegeben wurde. Vielleicht wird es später klarer, wenn wir diese Methoden nur aufrufen, wenn eine Litho vorhanden ist. Aktuell machen wir diese Prüfung erst innerhalb der jeweiligen Methoden.